### ISSUE TITLE
Check scaling sizes for auth map design

### ISSUE THREAD

**Issue by youngnick** [cilium/cilium#24617]
Spun out from cilium/design-cfps#5:

Check that the sizes for the fields in the auth map will scale correctly for large numbers of identities.

**Comment by youngnick** [comment]
From @joestringer:
> If we wanted to go whole-hog here then some diagrams could help - cases like:
>One local application connects to a single deployment that has two backends on two remote nodes. This results in four connections between two security identities:
Node A -> Node B for identity x->y (At Node A)
Node A -> Node B for identity x->y (At Node B)
Node A -> Node C for identity x->y (At Node A)
Node A -> Node C for identity x->y (At Node C)
Similarly, for one application that makes calls to two different remote Pods on other nodes with different identities, there would be four connections (even the same peer node) :
Node A -> Node B for identity x->y (At Node A)
Node A -> Node B for identity x->y (At Node B)
Node A -> Node B for identity x->z (At Node A)
Node A -> Node B for identity x->z (At Node B)

>We could probably explore even more cases, but this is probably enough info for now to consider the 'back of the hand' math on the impact.

So, a TODO is to consider how the number of rows will scale with number of identities and number of pods in each identity.

**Comment by youngnick** [comment]
Another item from @joestringer:
> (implementation details: this is probably somehow padded/aligned on some word boundary so common case for 3x__u32 is probably equivalent to 4x__u32, ie ~16B, so about 1MB as the last sentence implies)

> EDIT: Ah plus value, so maybe another half MB for that assuming <=8B of data.

So we need also to consider how big each row will be, and ensure we keep a reasonable table size.

**Comment by youngnick** [comment]
The last item will be to update the CFP in cilium/design-cfps with the details. (under the "Implementation Details" section inside the "Chosen alternative" section.

**Comment by youngnick** [comment]
We've set this to ~512k, the same size as the conntrack table, and because of how CNPs are namespaced, we expect the limiting factors to be the number of nodes in the cluster and the number of identities in use. This should be big enough for all but the largest installations, but we'll have to wait and see.

tl;dr we don't think there will be scaling issues for some time, if ever.

Remaining work here is to check if there is a metric available for the auth map pressure, since it's configurable once you know that.

**Comment by youngnick** [comment]
@sayboras was working on this in #26478, he will pick it up when he is back from leave.
